src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.c

```diff
@@ -743,6 +756,17 @@ static void rr_update_locked(grpc_exec_ctx *exec_ctx, grpc_lb_policy *policy,
       gpr_zalloc(sizeof(subchannel_data) * num_addrs);
   subchannel_list->num_subchannels = num_addrs;
   gpr_ref_init(&subchannel_list->refcount, 1);
+  if (p->latest_pending_subchannel_list != NULL && p->started_picking) {
+    if (GRPC_TRACER_ON(grpc_lb_round_robin_trace)) {
+      gpr_log(GPR_DEBUG,
+              "Shutting down latest pending subchannel list %p, about to be "
+              "replaced by newer latest %p",
+              (void *)p->latest_pending_subchannel_list,
+              (void *)subchannel_list);
+    }
+    rr_subchannel_list_shutdown(exec_ctx, p->latest_pending_subchannel_list,
```

Comment:
How is it safe to call this if we're not holding a ref on `p->latest_pending_subchannel_list`?

I think we need to hold a ref to both `p->subchannel_list` and `p->latest_pending_subchannel_list`.  Consequences of this:

- When promoting `p->latest_pending_subchannel_list` to `p->subchannel_list`, we should shutdown and unref `p->subchannel_list` and then simply reassign; there's no need to add a new ref to `p->latest_pending_subchannel_list` at that point.

- Any time we shut down a subchannel list, we should unref immediately after shutting down.  Given that, we might want to call `rr_subchannel_list_unref()` directly from `rr_subchannel_list_shutdown()` (and perhaps rename it to `rr_subchannel_list_shutdown_and_unref()`).